framework/src/modules/chain/synchronizer/block_synchronization_mechanism.js

```diff
@@ -138,53 +167,32 @@ class BlockSynchronizationMechanism {
 	 * @private
 	 */
 	async _requestLastCommonBlock(peer) {
-		const [lastBlock] = await this.storage.entities.Block.get({
-			sort: 'height:desc',
-			limit: 1,
-		});
-
 		const blocksPerRequestLimit = 10; // Maximum number of block IDs to be included in a single request
 		const requestLimit = 10; // Maximum number of requests to be made to the remote peer
 
 		let numberOfRequests = 0; // Keeps track of the number of requests made to the remote peer
-		let numberOfBlocks = 0; // Keeps track of the number of block IDs to be included in a request
-		let currentRound = Math.floor(
-			lastBlock.height / this.constants.activeDelegates,
-		); // Keeps track of the round number it is  being used to compute the first block of the round to be included in the request payload
 		let highestCommonBlock; // Holds the common block returned by the peer if found.
+		let currentRound = Math.floor(
+			this.blocks.lastBlock.height / this.constants.activeDelegates,
+		);
```

Comment:
Removed the comment here, intentionally?

Reply:
Added it back